### PR TITLE
refine pool create delete

### PR DESCRIPTION
**Pull request by jikui**
@<reviewer_id>
#### What issues does this address?
Fixes #<issueid>
WIP #<issueid>
...

### CODE REVIEW

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 1]
```diff
@@ -268,21 +267,29 @@ def _create(self, bigip, poolpayload, pool, service):
         if service.get('listener'):
             LOG.debug("Find a listener %s for create pool", listener)
             mgr = ListenerManager(self.driver)
-            old_listener = {}
-            old_listener['admin_state_up'] = listener['admin_state_up']
-            mgr.update(old_listener, listener, service)
+            mgr._search_element(listener, service)
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: you may not need to search it again.

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py [hunk 2]
```diff
@@ -268,21 +267,29 @@ def _create(self, bigip, poolpayload, pool, service):
...
         for listener in service['listeners']:
             if listener['default_pool_id'] == pool['id']:
                 service['listener'] = listener
-                old_listener = {}
-                old_listener['admin_state_up'] = listener['admin_state_up']
-                """ unmap the pool id and the listener"""
-                mgr.update(old_listener, listener, service)
-
+                """ Unmap listener and pool"""
+                mgr._search_element(listener, service)
```
Comment (zhaoqin-github) [inline f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: you may not need to search it again.

**Review by zhaoqin-github — CHANGES_REQUESTED** [review]

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py (hunk 1 above)
Comment (jikui) [reply f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: qin, if we call the pool's creation and deletion without listener's creation/deletion/update, then the listener's _search_element will not be used. Hence the service's listener key will not be set properly.
That's my thought about why I explicitly called _search_element for the listener here.

Let's talk in detail soon.

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py (hunk 2 above)
Comment (jikui) [reply f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: The same as above

File: f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py (hunk 1 above)
Comment (jikui) [reply f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py]: Qin is right.  Corrected it.

**Review by zhaoqin-github — APPROVED** [review]

**Pull request merged**